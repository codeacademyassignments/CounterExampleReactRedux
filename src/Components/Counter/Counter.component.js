import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button.component';

// class Counter extends Component{
// constructor(props){
//   super(props);
//   console.log('conso');
// }


// static getDerivedStateFromProps(props,state)
// {
//   console.log('done',props);
//   return state;
// }
// state = {
//   count: this.props.initialValue,
// }
// increment = ()=>{
//   this.setState({count: this.state.count+1});
// }
// decrement = ()=>{
//   console.log('minus');
//   this.setState({count:this.state.count-1});
// }
const Counter = ({ counterValue, incrementCounter, decrementCounter }) => (
  <div>
    <p className="counterValue">
Counter:
      {counterValue}
    </p>
    <div>
      <Button buttonText="ADD" onClick={incrementCounter} />
      <Button buttonText="MINUS" onClick={decrementCounter} />
    </div>
  </div>
);

Counter.propTypes = {
  counterValue: PropTypes.number.isRequired,
  incrementCounter: PropTypes.func.isRequired,
  decrementCounter: PropTypes.func.isRequired,
};
export default Counter;
