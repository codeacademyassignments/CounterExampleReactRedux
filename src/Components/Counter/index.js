import React , {Component} from 'react';
import Button from '../Button';

class Counter extends Component{

  state = {
    count: this.props.initialValue,
  }
  increment = ()=>{
    // console.log('somethign');
    // console.log(this.state.count);
    //this.state.count = this.state.count+1;
    this.setState({count: this.state.count+1});
  }
  decrement = ()=>{
    console.log('minus');
    this.setState({count:this.state.count-1});
  }
  render(){
    
    //const counter =10;
    return (
      <div>
        <p>Counter:{this.state.count}</p>
      <div>
      <Button buttonText = "ADD" onClick={this.increment}/>
      <Button buttonText = "MINUS" onClick={this.decrement}/>
</div>
</div>
    )
}
}

export default Counter