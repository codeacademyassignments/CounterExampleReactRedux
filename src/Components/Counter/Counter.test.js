import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './index';
import {shallow} from 'enzyme';

describe('Counter',()=>{
  it('should render a Counter',()=>{
    const tree = renderer.create(<Counter initialValue = {-10}/>).toJSON();
    expect(tree).toMatchSnapshot();

  })
  it('should increment counter on click',()=>{
    const wrapper = shallow(<Counter initialValue={0} />);
    wrapper.instance().increment();
    expect(wrapper.instance().state.count).toEqual(1); 
  })
  it('should decrement counter on click',()=>{
    const wrapper = shallow(<Counter initialValue={0} />);
    wrapper.instance().decrement();
    console.log('done',wrapper.instance().props.initialValue);
    expect(wrapper.instance().state.count).toEqual(-1); 
  })
  it('should get initial value whatever is passed',()=>{
    const wrapper = shallow(<Counter initialValue={0} />);
    expect(wrapper.instance().state.count).toEqual(0); 
  })
})