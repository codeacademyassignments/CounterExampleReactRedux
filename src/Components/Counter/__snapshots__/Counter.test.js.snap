// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a Counter 1`] = `
<div>
  <p>
    Counter:
    -10
  </p>
  <div>
    <button
      className="style"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="button"
    >
      ADD
    </button>
    <button
      className="style"
      onClick={[Function]}
      style={
        Object {
          "color": "black",
        }
      }
      type="button"
    >
      MINUS
    </button>
  </div>
</div>
`;
